sdk/data/azcosmos/cosmos_client.go

```diff
@@ -26,31 +30,52 @@ func (c *Client) Endpoint() string {
 	return c.endpoint
 }
 
-// NewClientWithKey creates a new instance of Cosmos client with the specified values. It uses the default pipeline configuration.
+// NewClientWithKey creates a new instance of Cosmos client with shared key authentication. It uses the default pipeline configuration.
 // endpoint - The cosmos service endpoint to use.
 // cred - The credential used to authenticate with the cosmos service.
 // options - Optional Cosmos client options.  Pass nil to accept default values.
 func NewClientWithKey(endpoint string, cred KeyCredential, o *ClientOptions) (*Client, error) {
-	return &Client{endpoint: endpoint, pipeline: newPipeline(cred, o)}, nil
+	return &Client{endpoint: endpoint, pipeline: newPipeline([]policy.Policy{newSharedKeyCredPolicy(cred)}, o)}, nil
 }
 
-func newPipeline(cred KeyCredential, options *ClientOptions) azruntime.Pipeline {
+// NewClient creates a new instance of Cosmos client with Azure AD access token authentication. It uses the default pipeline configuration.
+// endpoint - The cosmos service endpoint to use.
+// cred - The credential used to authenticate with the cosmos service.
+// options - Optional Cosmos client options.  Pass nil to accept default values.
+func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (*Client, error) {
+	scope, err := createScopeFromEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{endpoint: endpoint, pipeline: newPipeline([]policy.Policy{runtime.NewBearerTokenPolicy(cred, scope, nil), &cosmosBearerTokenPolicy{}}, o)}, nil
```

Comment:
so we have a bearerTokenPolicy to append the authorization header then cosmosBearerTokenPolicy to read and modify the token?

I think it's cleaner if we bypass BearerTokenPolicy, maybe feed azcore.TokenCredential direct to cosmosBearerTokenPolicy, and generate the correct format?

Reply:
We discussed this with @jhendrixMSFT - currently there is no way for us to pass the format to the BearerTokenPolicy. Either the BearerTokenPolicy format is configurable or the ExpiringResource is made public. I can remove or change the code once a decision is made on that regard, meanwhile, this approach is the only alternative.

Reply:
thanks, that's pretty good explanation.  you can resolve this comment.